src/main/java/book/shop/service/impl/BookServiceImpl.java

```diff
@@ -11,14 +15,36 @@
 @RequiredArgsConstructor
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
 
     @Override
-    public Book save(Book book) {
-        return bookRepository.save(book);
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toModel(requestDto);
+        return bookMapper.toDto(bookRepository.save(book));
     }
 
     @Override
-    public List<Book> findAll() {
-        return bookRepository.findAll();
+    public BookDto getBookById(Long id) {
+        Book book = bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find the book with id: " + id)
+        );
+        return bookMapper.toDto(book);
+    }
+
+    @Override
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public BookDto updateBookById(Long id) {
+        return bookRepository.updateBookById(id);
```

Comment:
```suggestion
        Book book = bookRepository.findById(id)
                .orElseThrow(() -> new EntityNotFoundException.......
        bookMapper.updateBookFromDto(bookDto, book);
        return bookMapper.toDto(bookRepository.save(book));
```

Create updateBookFromDto in BookMapper